presto-common/src/main/java/com/facebook/presto/common/type/TypeSignature.java

```diff
@@ -146,6 +146,25 @@ public List<TypeSignature> getTypeParametersAsTypeSignatures()
         return result;
     }
 
+    public List<TypeSignature> getTypeOrNamedTypeParametersAsTypeSignatures()
```

Comment:
Can you instead modify getTypeParametersAsTypeSignatures. I don't think there's any reason NamedTypeParameters are excluded there.